### PR TITLE
refactor gas-fee-display component

### PR DESCRIPTION
**Pull request by TrejGun**
#4622

### CODE REVIEW

**Comment by TrejGun** [comment]
I don't uderstad why it can't find button by text
there where no text before
https://github.com/MetaMask/metamask-extension/blob/develop/ui/app/components/send/gas-fee-display-v2.js#L49

**Comment by danjm** [comment]
Thanks @TrejGun! I will review and also provide some help on getting those tests passing within the next 24 hours.

**Comment by danjm** [comment]
Code looks good @TrejGun ! And nice job on the unit tests. I'll investigate why e2e and integration tests are failing.

**Review by danjm — APPROVED** [review]

File: ui/app/components/pending-tx/confirm-send-ether.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const {
   calcGasTotal,
   isBalanceSufficient,
 } = require('../send_/send.utils')
-const GasFeeDisplay = require('../send/gas-fee-display-v2')
+const GasFeeDisplay = require('../send_/send-content/send-gas-row/gas-fee-display/gas-fee-display.component')
```
Comment (danjm) [inline ui/app/components/pending-tx/confirm-send-ether.js]: Needs to be `const GasFeeDisplay = require('../send_/send-content/send-gas-row/gas-fee-display/gas-fee-display.component').default`

Read here for explanation: https://stackoverflow.com/questions/33704714/cant-require-default-export-value-in-babel-6-x

File: ui/app/components/pending-tx/confirm-send-token.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ abiDecoder.addABI(tokenAbi)
 const actions = require('../../actions')
 const clone = require('clone')
 const Identicon = require('../identicon')
-const GasFeeDisplay = require('../send/gas-fee-display-v2.js')
+const GasFeeDisplay = require('../send_/send-content/send-gas-row/gas-fee-display/gas-fee-display.component.js')
```
Comment (danjm) [inline ui/app/components/pending-tx/confirm-send-token.js]: Needs to be `const GasFeeDisplay = require('../send_/send-content/send-gas-row/gas-fee-display/gas-fee-display.component').default`

Read here for explanation: https://stackoverflow.com/questions/33704714/cant-require-default-export-value-in-babel-6-x

**Review by danjm — CHANGES_REQUESTED** [review]
@TrejGun just made two comments on the code. I think if you fix that, the tests will pass.

**Comment by TrejGun** [comment]
oh, my bad, fixed

**Comment by TrejGun** [comment]
what do you think about making `contextTypes` a static prop same as `propTypes`?

**Review by danjm — APPROVED** [review]

**Comment by danjm** [comment]
> what do you think about making contextTypes a static prop same as propTypes

I think that is a good idea. Do you want to do it in a separate PR?

**Comment by TrejGun** [comment]
@danjm i do

**Comment by TrejGun** [comment]
@danjm #4702

**Pull request merged**